recipe-server/normandy/recipes/api/v2/serializers.py

```diff
@@ -154,7 +154,17 @@ def validate_extra_filter_expression(self, value):
         jexl.add_transform('preferenceIsUserSet', lambda x: x)
         jexl.add_transform('preferenceExists', lambda x: x)
 
-        errors = list(jexl.validate(value))
+        try:
+            errors = list(jexl.validate(value))
+        except Exception as exc:
+            # The JEXL parser can occasionally throw exceptions when
+            # called to validate certain invalid inputs.
+            # Catch them and at least indicate the field that failed,
+            # even if we can't explain exactly what the problem was.
+            # See https://github.com/mozilla/normandy/issues/1059.
+            error = f'The JEXL parser failed to validate {value}'
```

Comment:
I know this is merged but my python-code-smell-alarm buzzed when I saw this. 
I guess there isn't a way to re-wrap standard exception classes into specific ones inside `jexl`. 

Also the variable `exc` doesn't get used. It might be worth calling it's str method as part of the `error` string. E.g.
```python
error = f'The JEXL parser failed to validate {value} ({exc})'
```

Reply:
That's actually a good point, @peterbe. I probably should have spotted `except Exception`, and our linter should have caught the unused `exc`.

Including the exception in the output seems a bit problematic, as it may expose server state. Ultimately, I think patching upstream, like @glasserc did, is the correct response.

Maybe we should narrow the scope of this to only catch the specific error we saw in #1059? So it would be `except jexl.VisitationError`. That could allow other errors to slip out and cause server 500s, but it is more hygienic. Thoughts @peterbe, @glasserc?

Reply:
The reason the code smells bad is because we are trying to work around a bug rather than actually fixing it. And not even just one specific bug, but a hypothetical future bug that we expect the `pyjexl` library to manifest, which we can't point to but only imagine may come about based on our experience using this library. That bug, just to be explicit, is that the `validate()` function may throw an exception rather than generate a list of errors in the input. This would be a violation of the contract that the `validate()` function has. Because it would be a violation of the contract, and again because we have no idea what such a violation would look like, it's impossible to write clean code that addresses it. We have no reason to suspect that this future bug would even raise the same `jexl.VisitationError` instead of e.g. `KeyError` or `AttributeError`. A bug in a library could raise essentially any error. As indicated in PEP8, the way to catch "any exception that signifies an error" is to catch `Exception`.

If the bad-smelling code bothers you, I'd recommend reverting the PR. We could clean it up a tiny bit (removing the unused variable), but again, it can't really address the problem, just paper over it. Instead, a better use of time might be to try to improve the robustness of the `jexl` library so that it doesn't ever have this kind of bug. For example, I have opened https://github.com/Osmose/pyjexl/pull/13 as a first step in this direction.

Reply:
Is it so bad to have 500s? We have Sentry and stuff. 
But I'm unsure what the context of all of this is. 
Personally I like @glasserc where you instead take this as an indicator that it's time to go deep down inside the `.validate` method instead. If you pass in invalid data to a function called `validate` it should probably not crash and return false. 

Perhaps, as a first step is to focus on the inner problem that it can raise errors. 
Then secondly, write down specific pyjexl'esque exceptions that *might* happen and shouldn't wake people up at night.